Content.Shared/_CM14/Scoping/SharedScopeSystem.cs

```diff
@@ -74,115 +80,92 @@ private void OnToggleAction(Entity<ScopeComponent> ent, ref ToggleActionEvent ar
+        if (ent.Comp.ScopingDirection != dir)
+            StopScoping(ent);
     }
 
-    public bool StartScoping(EntityUid item, ScopeComponent component, EntityUid user)
+    private void StartScoping(Entity<ScopeComponent> scope, EntityUid user)
     {
-        if (component.IsScoping)
-            return false;
-
-        if (TryComp(user, out ScopeUserComponent? scopeUserComp))
-        {
-            scopeUserComp.ScopingItem = item;
-            Dirty(user, scopeUserComp);
-        }
-
-        var msgUser = Loc.GetString("cm-action-popup-scoping-user", ("scope", Name(item)));
-        _popupSystem.PopupClient(msgUser, user, user);
+        var xform = Transform(user);
+        var cardinalDir = xform.LocalRotation.GetCardinalDir();
+        scope.Comp.User = user;
+        scope.Comp.ScopingDirection = cardinalDir;
 
-        _actionsSystem.SetToggled(component.ScopingToggleActionEntity, true);
+        Dirty(scope);
+        var scoping = EnsureComp<ScopingComponent>(user);
+        scoping.Scope = scope;
 
-        StartScopingCamera(user, component);
+        var targetOffset = cardinalDir.ToVec() * ((SmallestViewpointSize * scope.Comp.Zoom - 1) / 2);
+        scoping.EyeOffset = targetOffset;
 
-        component.IsScoping = true;
-        Dirty(item, component);
+        var msgUser = Loc.GetString("cm-action-popup-scoping-user", ("scope", Name(scope.Owner)));
+        _popup.PopupClient(msgUser, user, user);
 
-        return true;
+        _actionsSystem.SetToggled(scope.Comp.ScopingToggleActionEntity, true);
+        _contentEye.SetZoom(user, Vector2.One * scope.Comp.Zoom, true);
     }
 
-    public bool StopScoping(EntityUid item, ScopeComponent component, EntityUid user)
+    private void StopScoping(Entity<ScopeComponent> scope)
     {
-        if (!component.IsScoping)
-            return false;
-
-        if (TryComp(user, out ScopeUserComponent? scopeUserComp))
-        {
-            scopeUserComp.ScopingItem = null;
-            Dirty(user, scopeUserComp);
-        }
-
-        var msgUser = Loc.GetString("cm-action-popup-scoping-stopping-user", ("scope", Name(item)));
-        _popupSystem.PopupClient(msgUser, user, user);
+        if (scope.Comp.User is not { } user)
+            return;
 
-        _actionsSystem.SetToggled(component.ScopingToggleActionEntity, false);
+        RemCompDeferred<ScopingComponent>(user);
 
-        StopScopingCamera(user, component);
+        scope.Comp.User = null;
+        scope.Comp.ScopingDirection = null;
+        Dirty(scope);
 
-        component.IsScoping = false;
-        Dirty(item, component);
+        var msgUser = Loc.GetString("cm-action-popup-scoping-stopping-user", ("scope", Name(scope.Owner)));
+        _popup.PopupClient(msgUser, user, user);
 
-        return true;
+        _actionsSystem.SetToggled(scope.Comp.ScopingToggleActionEntity, false);
+        _contentEye.ResetZoom(user);
+        _contentEye.SetZoom(user, Vector2.One * scope.Comp.Zoom, true);
```

Comment:
`_contentEye.ResetZoom(user);` should be enough.
```suggestion
```

Reply:
Oops, fixed in [`c91a0f5` (#2426)](https://github.com/CM-14/CM-14/pull/2426/commits/c91a0f51f44bfce6a20a2494f2bedfa1c28b48dd)